packages/design-system/src/styles/_base.scss

```diff
@@ -261,55 +261,56 @@ $breakpoints: (
   color: var(--color-muted-inverse);
 }
 
-// @TODO: Remove the semantic classes h1, h2, h3, display, title, etc... after deprecating
+// WNMGDS-2146: Deprecate semantic font-size classes
+// Refer to `_set-for-deprecation.scss` for complete list
+// Classes marked with 💀 are deprecated and will be removed in the next major release
 
-.ds-display,
+.ds-display, // 💀
 .ds-text-heading--5xl {
   @include heading--5xl;
 }
 
-.ds-title,
+.ds-title, // 💀
 .ds-text-heading--4xl {
   @include heading--4xl;
 }
 
-.ds-h1,
+.ds-h1, // 💀
 .ds-text-heading--3xl {
   @include heading--3xl;
 }
 
-.ds-h2,
+.ds-h2, // 💀
 .ds-text-heading--2xl {
   @include heading--2xl;
 }
 
-.ds-h3,
+.ds-h3, // 💀
 .ds-text-heading--xl {
   @include heading--xl;
 }
 
-.ds-h4,
+.ds-h4, // 💀
 .ds-text-heading--lg {
   @include heading--lg;
 }
 
-.ds-h5,
+.ds-h5, // 💀
 .ds-text-heading--md {
   @include heading--md;
 }
 
-.ds-h6,
+.ds-h6, // 💀
 .ds-text-heading--sm {
   @include heading--sm;
-}
 
-.ds-base--inverse .ds-h6,
-.ds-base--inverse .ds-text-heading--sm {
-  color: var(--color-muted-inverse);
+  .ds-base--inverse & {
```

Comment:
this is more of a cleanup thing - we duplicated the .ds-h6 and .ds-text-heading--sm classes a few times and this consolidates things